include/nuttx/compiler.h

```diff
@@ -107,6 +107,10 @@
 
 #  define aligned_data(n) __attribute__ ((aligned(n)))
 
+/* Data location */
+
+#  define locate_data(n) __attribute__ ((section(n)))
```

Comment:
Since __attribute__ ((section(n))) can also apply to function, could we give a more general name here?

Reply:
sure or we can add `locate_code` and keep the abstraction

Reply:
good!

Reply:
@xiaoxiang781216 - done.

Reply:
LGTM.